cli/cmd/inject_test.go

```diff
@@ -46,3 +46,48 @@ func TestInjectYAML(t *testing.T) {
 		})
 	}
 }
+
+func TestRunInjectCmd(t *testing.T) {
+	t.Run("Do not print invalid YAML on inject error", func(t *testing.T) {
+		errBuffer := &bytes.Buffer{}
+		outBuffer := &bytes.Buffer{}
+		expectedErrorMsg := `Error injecting conduit proxy: error converting YAML to JSON: yaml: line 14: did not find expected key`
+
+		in, err := os.Open("testdata/inject_gettest_deployment.input.yml")
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		errCode := runInjectCmd(in, errBuffer, outBuffer)
+
+		if errCode != 1 {
+			t.Fatalf("Expected error code to be 1 but got: %d", errCode)
+		}
+
+		if len(outBuffer.Bytes()) != 0 {
+			t.Fatalf("Expected output buffer to be empty but got: \n%v", outBuffer)
+		}
+
+		actualErrorMsg := errBuffer.String()
+		diffCompare(t, actualErrorMsg, expectedErrorMsg)
+	})
+	t.Run("Do not print invalid YAML on inject error", func(t *testing.T) {
```

Comment:
The string passed to t.Run() is not right for this test case.

More generally, it's better to write these tests as table-driven tests since they are much easier to review and expand upon in future work:

```go
    testCases := []struct {
 		inputFileName  string
 		stdoutFileName string
 		stderrGoldenFileName string
		exitCode int
 	}{
 		{ <inputs/outputs for the passing test case> },
 		{ <inputs/outputs for the failing  test case> },
 	}
	
	for i, tc := range(testCases) {
	    t.Run(...) {
		    // Your test code here.
		}
	}
```

(I didn't `go fmt` this.)